docs/testing/snapshot-testing.md

```diff
@@ -0,0 +1,129 @@
+# Snapshot testing
+
+This is an overview of [snapshot testing](https://facebook.github.io/jest/docs/en/snapshot-testing.html) and how to best use snapshot tests in Calypso.
+
+## Broken snapshots
+
+When a snapshot test fails, it just means that a component's rendering has changed. If that was unintended,  then the snapshot test just prevented a bug 😊
+
+However, if the change was intentional, follow these steps to update the snapshot:
+
+1. Run the following to update the snapshots:
+   ```sh
+   # --testPathPattern is optional but will be much faster by only running matching tests
+   npm run test-client -- --updateSnapshot --testPathPattern client/components
+   ```
+1. Review the diff and ensure the changes are expecte and intentional
+1. Commit
+
+## What are snapshots?
+
+Snapshots are just a representation of some data structure generated in our tests. Snapshots are stored in files and committed alongside the tests. When the tests are run, the data structure generated is compared with the snapshot on file.
+
+It's very easy to make a snapshot:
+
+```js
+test( 'foobar test', () => {
+  const foobar = { foo: 'bar' };
+  expect( foobar ).toMatchSnapshot();
+} );
+```
+
+This is the produced snapshot:
+
+```js
+exports[`test foobar test 1`] = `
+  Object {
+    "foo": "bar",
+  }
+`;
+```
+
+You should never interact with the snapshot directly, they are generated by tests. However you can see that it's a representation the test data.
+
+## Advantages
+
+
+* Trivial and concise to write tests
+* Protect against unintentional changes
+* Simple to work with
```

Comment:
I figured out that it is an amazing way to see the representation of the component without loading the browser.

Reply:
* Representation of the component without needing the DOM

Something like that?

Reply:
Maybe broader:

> Reveal the internal structure without need to run the application